### PR TITLE
fix: concurrent access + outdated statuses

### PR DESCRIPTION
**Pull request by gfyrag**

### CODE REVIEW

File: ee/agent/internal/informer_stacks.go [hunk 1]
```diff
@@ -115,7 +187,8 @@ func NewStackEventHandler(logger sharedlogging.Logger, membershipClient Membersh
 				logger.Errorf("Unable to send stack delete to server: %s", err)
 			}
 
-			if _, ok := stacks[stack.GetName()]; !ok {
+			modules, ok := stacks.Get(stack.GetName())
+			if !ok {
 				logger.Debugf("Stack '%s' not initialized in memory", stack.GetName())
 				return
 			}
```
Comment (Dav-14) [inline ee/agent/internal/informer_stacks.go:194]: Missing on push here ?

File: ee/agent/internal/informer_stacks.go [hunk 2]
```diff
@@ -52,7 +113,7 @@ func NewStackEventHandler(logger sharedlogging.Logger, membershipClient Membersh
 		if err != nil {
 			panic(err)
 		}
-		for _, module := range GetExpectedModules(stack.GetName(), stacks) {
+		for _, module := range modules {
 			if !slices.Contains(modules, module) {
 				return generated.StackStatus_Progressing
 			}
```
Comment (Dav-14) [inline ee/agent/internal/informer_stacks.go:119]: The modules should be based from inmemories and not on the one "actually attached" to stack CRD

File: ee/agent/internal/informer_stacks.go [hunk 3]
```diff
@@ -115,7 +188,8 @@ func NewStackEventHandler(logger sharedlogging.Logger, membershipClient Membersh
 				logger.Errorf("Unable to send stack delete to server: %s", err)
 			}
 
-			if _, ok := stacks[stack.GetName()]; !ok {
+			modules, ok := stacks.Get(stack.GetName())
+			if !ok {
 				logger.Debugf("Stack '%s' not initialized in memory", stack.GetName())
 				return
```
Comment (Dav-14) [inline ee/agent/internal/informer_stacks.go:194]: missing onPush ?

File: ee/agent/internal/informer_stacks.go [hunk 4]
```diff
@@ -10,17 +13,76 @@ import (
...
+		stacks: make(map[string][]string),
+		onPush: make(map[string]func([]string)),
+	}
 }
 
-func NewStackEventHandler(logger sharedlogging.Logger, membershipClient MembershipClient, stacks InMemoryStacksModules) cache.ResourceEventHandlerFuncs {
-	sendStatus := func(interpretedStatus generated.StackStatus, stackName string, status *structpb.Struct) {
+func NewStackEventHandler(logger sharedlogging.Logger, membershipClient MembershipClient, stacks *InMemoryStacksModules) cache.ResourceEventHandlerFuncs {
+	sendStatus := func(interpretedStatus generated.StackStatus, stackName string, status *structpb.Struct, modules []string) {
+		logger.Infof("Send status %s for stack %s", interpretedStatus, stackName)
```
Comment (Dav-14) [inline ee/agent/internal/informer_stacks.go:85]: `modules` not needed in sendStatus func

File: ee/agent/internal/informer_stacks.go (hunk 2 above)
Comment (gfyrag) [reply ee/agent/internal/informer_stacks.go:120]: It is the case

**Review by Dav-14 — APPROVED** [review]

**Pull request merged**